zerver/fixtures/markdown_test_cases.json

```diff
@@ -801,12 +806,12 @@
     ],
     [
       "javascript://foo.com",
-      "<p>javascript://foo.com</p>",
+      "<p>javascript://<a href=\"http://foo.com\" target=\"_blank\" title=\"http://foo.com\">foo.com</a></p>",
```

Comment:
hm .. I think the original behavior was the desired one; we only want to be generous in the explicit `https://` case.

Reply:
I think this is preferable. Given foo.com is still a URL, and should be rightly marked. Your suggestions on it?

Reply:
not all url-like things are over http, though. E.g. ftp://some.uri is a valid thing, that isn't over http. I'm also worried that companies with internal networks might use some weird blah://internal.com type scheme, that this would http-ify.

@showell, thoughts?

Reply:
The frontend behavior is to not linkify the `foo.com` in `javascript://foo.com`.

Reply:
Another data point: `mailto:example@example.com` is linkified/href-ed in the backend (without the mailto being part of the link), but not in the frontend.